src/overworld.c

```diff
@@ -409,6 +409,14 @@ void Overworld_ResetBattleFlagsAndVars(void)
         VarSet(B_VAR_TERRAIN_TIMER, 0);
     #endif
 
+    #if B_SET_TRICK_ROOM
+        FlagClear(B_SET_TRICK_ROOM);
```

Comment:
does this reset on whiteout too?

Reply:
it should but I can test.

Reply:
It only resets on whiteout otherwise as long as the flag is set each battle will have Trick Room. I'm not sure what the best way to handle this would be. I assume if someone wants a perma trick room battle they would want it to be permanent across multiply battles.

Reply:
that seems like the correct behavior, yeah